app/http.php

```diff
@@ -0,0 +1,366 @@
+
+        $log->addExtra('file', $error->getFile());
+        $log->addExtra('line', $error->getLine());
+        $log->addExtra('trace', $error->getTraceAsString());
+        $log->addExtra('detailedTrace', $error->getTrace());
+
+        $log->setAction($action);
+
+        $isProduction = App::getEnv('OPEN_RUNTIMES_PROXY_ENV', 'development') === 'production';
+        $log->setEnvironment($isProduction ? Log::ENVIRONMENT_PRODUCTION : Log::ENVIRONMENT_STAGING);
+
+        $responseCode = $logger->addLog($log);
+        Console::info('Proxy log pushed with status code: ' . $responseCode);
+    }
+
+    Console::error('[Error] Type: ' . get_class($error));
+    Console::error('[Error] Message: ' . $error->getMessage());
+    Console::error('[Error] File: ' . $error->getFile());
+    Console::error('[Error] Line: ' . $error->getLine());
+}
+
+Console::success('Waiting for executors to start...');
+
+$startupDelay = \intval(App::getEnv('OPEN_RUNTIMES_PROXY_STARTUP_DELAY', '0'));
+if ($startupDelay > 0) {
+    \sleep((int) ($startupDelay / 1000));
+}
+
+\go(function () use ($register) {
+    if (App::getEnv('OPEN_RUNTIMES_PROXY_HEALTHCHECK', 'enabled') === 'enabled') {
+        fetchExecutorsState(true);
+    } else {
+        $executorStates = $register->get('executorStates');
+
+        // If no health check, mark all as online
+        $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+        foreach ($executors as $executor) {
+            $executorStates->set($executor, [
+                'status' => 'online',
+                'hostname' => $executor,
+                'state' =>  \json_encode([])
+            ]);
+        }
+    }
+});
+
+Console::log('State of executors at startup:');
+
+go(function () use ($register) {
+    $executorStates = $register->get('executorStates');
+
+    $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    foreach ($executors as $executor) {
+        $state = $executorStates->exists($executor) ? $executorStates->get($executor) : null;
+
+        if ($state === null) {
+            Console::warning('Executor ' . $executor . ' has unknown state.');
+        } else {
+            Console::log('Executor ' . $executor . ' is ' . ($state['status'] ?? 'unknown') . '.');
+        }
+    }
+});
+
+Swoole\Event::wait();
+
+// Wildcard action
+App::error()
+    ->inject('request')
+    ->inject('response')
+    ->inject('roundRobinAtomic')
+    ->inject('executorStates')
+    ->action(function (Request $request, Response $response, Atomic $roundRobinAtomic, Table $executorStates) {
+        $secretKey = \explode(' ', $request->getHeader('authorization', ''))[1] ?? '';
+
+        if (empty($secretKey)) {
+            throw new Exception('Incorrect proxy key.', 401);
+        }
+        if ($secretKey !== App::getEnv('OPEN_RUNTIMES_PROXY_SECRET', '')) {
+            throw new Exception('Incorrect proxy key.', 401);
+        }
```

Comment:
As a best practice, and in case we will add more routes here, we should set authentication as part of an init hook. If this doesn't work with the way we use the error handler as a wildcard, we should add a comment and address it after we add the wildcard support as a feature in `utopia/framework`

Reply:
Moved to init hook.